src/app/api/http-courses.service.ts

```diff
@@ -0,0 +1,35 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { Course, RequestBody } from '@app/models';
+
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HttpCoursesService {
+  url: string = 'http://localhost:3004';
```

Comment:
do not hardcode url in two places. Move it to some common constant so you could change it only in one place